src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -861,7 +861,7 @@ ASTPtr MutationsInterpreter::prepareInterpreterSelectQuery(std::vector<Stage> &
         {
             if (!actions_chain.steps.empty())
                 actions_chain.addStep();
-            stage.analyzer->appendExpression(actions_chain, ast, dry_run);
+            stage.analyzer->appendExpression(actions_chain, ast, dry_run, true /* disallow_arrayJoin */);
```

Comment:
I would prefer not to add a new flag to ExpressionAnalyzer.
There is a method `ActionsDAG::hasArrayJoin`.
I would rather add this check to `MutationsInterpreter::addStreamsForLaterStages`. There is an inner loop where we iterate over all stages, we can add a check there.

Reply:
this approach is indeed better. addressed it, thanks.